src/ScottPlot4/ScottPlot/Statistics/Histogram.cs

```diff
@@ -1,59 +1,197 @@
-    public readonly int[] Counts;
+    public readonly double[] Counts;
 
     /// <summary>
-    /// Lower edge for each bin
+    /// Running total of all values counted.
     /// </summary>
-    public readonly double[] BinEdges;
+    public double Sum { get; private set; }
 
     /// <summary>
-    /// Number of values that were smaller than the lower edge of the smallest bin
+    /// Total number of values accumulated.
     /// </summary>
-    public int MinOutlierCount { get; private set; } = 0;
+    public int ValuesCounted { get; private set; }
 
     /// <summary>
-    /// Number of values that were greater than the upper edge of the smallest bin
+    /// Lower edge for each bin.
     /// </summary>
-    public int MaxOutlierCount { get; private set; } = 0;
+    public readonly double[] BinEdges;
+
+    /// <summary>
+    /// Number of bins.
+    /// </summary>
+    public readonly int BinCount;
 
     /// <summary>
     /// Default behavior is that outlier values are not counted.
     /// If this is enabled, min/max outliers will be counted in the first/last bin.
     /// </summary>
-    public bool AddOutliersToEdgeBins { get; set; } = false;
+    public readonly bool AddOutliersToEdgeBins;
 
     /// <summary>
-    /// Create a histogram which will count values supplied by <see cref="Add"/> and <see cref="AddRange"/>
+    /// Lower edge of the first bin
     /// </summary>
-    public Histogram(double firstBin, double binSize, int binCount)
+    public readonly double Min;
+
+    /// <summary>
+    /// Upper edge of the last bin
+    /// </summary>
+    public readonly double Max;
+
+    /// <summary>
+    /// The calculated bin size.
+    /// </summary>
+    public double BinSize { get; private set; } = 0;
+
+    /// <summary>
+    /// Number of values that were smaller than the lower edge of the first bin.
+    /// </summary>
+    public int MinOutlierCount { get; private set; } = 0;
+
+    /// <summary>
+    /// Number of values that were greater than the upper edge of the last bin.
+    /// </summary>
+    public int MaxOutlierCount { get; private set; } = 0;
+
+    /// <summary>
+    /// Create a histogram which will count values supplied by <see cref="Add(double)"/> and <see cref="AddRange"/>
+    /// </summary>
+    public Histogram(double min, double max, int binCount, bool addOutliersToEdgeBins = false)
     {
-        Counts = new int[binCount];
+        Min = min;
+        Max = max;
+        BinCount = binCount;
+        AddOutliersToEdgeBins = addOutliersToEdgeBins;
+        Counts = new double[binCount];
         BinEdges = new double[binCount];
 
+        // create evenly sized bins
+        BinSize = (Max - Min) / binCount;
+
         for (int i = 0; i < binCount; i++)
         {
-            BinEdges[i] = firstBin + binSize * i;
+            BinEdges[i] = min + BinSize * i;
+        }
+    }
+
+    public static Histogram WithFixedSizeBins(double min, double max, double binSize, bool addOutliersToEdgeBins = false)
```

Comment:
@bukkideme to clear up any confusion, this is why I mentioned #2299

Reply:
Hi @Xerxes004, thanks for reminding us about #2299! It would be great to solve that issue with this PR. I'll follow-up with a comment on this PR in a few minutes when I learn more about that issue...

Reply:
Fixed by https://github.com/ScottPlot/ScottPlot/pull/2403/commits/43d0664729ba20a3658af97e739c43a20579f74a 🚀